File: Algorithms/Implementation/Cut the sticks/sticks.js
'use strict';

process.stdin.setEncoding('utf-8');
var os = require('os');
var input = '';

process.stdin.on('data', data => {
	input += data;

	if(data.split(os.EOL)[0] == '') {
		process.stdin.emit('end');
	}
});

process.stdin.on("end", _ => {
	main();
});

function main() {
	var n = parseInt(input.split(os.EOL)[0]);
	var arr = input.split(os.EOL)[1].split(' ').map(x => +x);

	let result = cutTheSticks(arr);

	for (var s of result) {
		console.log(s);
	}
}

function cutTheSticks(arr) {
	arr.sort((a, b) => { return a - b;});
	var result = Array(1).fill(arr.length);

	for (var i = 0; i < arr.length-1; i++) {
		if (arr[i] != arr[i+1]) {
			result.push(arr.length - (i + 1));
		}
	}

	return result;
}
			

File: Algorithms/Implementation/Cut the sticks/Sticks.java
import java.util.Arrays;
import java.util.Scanner;

class Sticks {
	static int[] cutTheSticks(int[] arr) {
		Arrays.sort(arr);
		int[] result = {arr.length};

		for (int i = 0; i < arr.length - 1; i++) {
			if (arr[i] != arr[i+1]) {
				int[] temp = new int[result.length + 1];

				for (int j = 0; j < result.length; j++) {
					temp[j] = result[j];
				}
				temp[result.length] = arr.length - (i + 1);
				result = temp;
			}
		}

		return result;
	}

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
		int n = in.nextInt();
		int[] arr = new int[n];
		in.skip("\n");

		for (int i = 0; i < n; i++) {
			arr[i] = in.nextInt();
		}

		int[] result = cutTheSticks(arr);

		for (int i = 0; i < result.length; i++) {
			System.out.println(result[i]);
		}
	}
}